### PR TITLE
feat(specs): add executor economics page

### PR DESCRIPTION
**Pull request by smartcontracts**
Adds a page to the specs folder for executor economics.

### CODE REVIEW

**Comment by sam-goldman** [comment]
Looks great! Just had questions on a few points.

File: specs/executor-economics.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    bytes32 _bundleId
+  ) external view returns (address, uint256);
+}
+```
+
+Each ESS then implements its own logic for selecting the executor for a given project and upgrade bundle. A project’s ESS cannot be changed while an upgrade is active. Anyone may create a new ESS at any time.
+
+#### ESS #1: Simple Lock
+
+Our first MVP ESS is the `SimpleLockESS`. Executors may post a bond of `executor_bond_amount` to claim the sole right to execute actions for the locked upgrade for a period of `upgrade_lock_time`. Only the first executor to post a bond gains this right. If the selected executor completes the upgrade within the `upgrade_lock_time` or if the upgrade is cancelled, the bond is returned. If the executor fails to complete the upgrade within the `upgrade_lock_time`, a new executor may post a bond instead and the original bond is forfeited and transferred into the wallet of the project’s `ChugSplashManager` contract. We note that this creates a PGA for the transaction that posts the bond and locks the upgrade but removes PGAs during the execution process.
```
Comment (yorhodes) [inline specs/executor-economics.md:52]: > the selected executor completes the upgrade

how is this criteria determined? some bytecode checks onchain?

**Review by yorhodes — COMMENTED** [review]
some questions I had when reading

File: specs/executor-economics.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+### Withdrawal Locks
+
+Funds cannot be withdrawn while an upgrade is active. This guarantees that executors will be paid for any transactions submitted during the upgrade. Users may withdraw funds after an upgrade is completed, reverted, paused, or cancelled. However, to prevent users from trolling executors by immediately cancelling and withdrawing funds, users must post a bond of `user_bond_amount` to be forfeited if an upgrade is cancelled. Upgrade cancellation is a potentially dangerous action and should not be executed unless in an emergency. We therefore do not expect cancellations (and bond forfeits) to occur frequently.
+
+### Executor Selection Strategies
+
+If upgrade execution were to be a free-for-all, we would expect to see Priority Gas Auctions (PGAs) to develop where executors aim to guarantee that their execution transactions land on-chain before the transactions of any other executors. PGAs generally drive up the cost of upgrades and cause executors to pay transaction fees that they cannot be rewarded for. Furthermore, the cost of a PGA is directed to block producers which means that economic value is leaving the ChugSplash system.
```
[the remark this answers is not in the document]
Comment (smartcontracts) [reply specs/executor-economics.md:24]: Main issue is that the executors need to download data off-chain, run some code, then figure out the specific inputs to execute the transactions. I'm not sure if any keeper networks support this sort of complex stuff, but I agree it's worth looking into before we go and do all this work ourselves.

File: specs/executor-economics.md (hunk 1 above)
Comment (smartcontracts) [reply specs/executor-economics.md:52]: We keep track of the number of executed steps vs the number of total steps. So as soon as executed = total, we say the upgrade is finished.

**Pull request merged**